### PR TITLE
stop all the empty string answers

### PR DESCRIPTION
**Pull request by andrewroth**
@twinge @jbirdjavi because the initial answers on the element were set using
responses(answer_sheet), that filters out empty strings.  So any update
where people hadn't chosen a dropdown value it was writing an empty
string.  You might want to check your database and see if there are a
lot of extra empty strings if you depend on pulling out answers
manually, like we are for apptrack in one-app/cap.  Using
display_response works because it already filters out empty strings

We could add where value != "" everywhere in our apptrack code but this seems like the better way to do it.  Also cleaned up the code in a few other places.

### CODE REVIEW

**Comment by andrewroth** [comment]
Ok, that's weird that it's failing on Asset was not declared to be precompiled in production... seems totally unrelated to what I was doing

**Comment by twinge** [comment]
which part of this change stops the empty strings?

File: app/models/fe/question.rb [hunk 1]
```diff
@@ -186,7 +186,7 @@ def set_response(values, answer_sheet)
         #   raise object_name.inspect + ' == ' + attribute_name.inspect
         # end
       else
-        @answers ||= []
+        @answers ||= sheet_answers.to_a
```
Comment (andrewroth) [inline app/models/fe/question.rb]: @twinge this is what stops the empty string answers.  Note that I moved this from the question_set line "-        question.answers = question.responses(answer_sheet) #answers[question.id]" into here.  Seemed to make more sense to me that a question can set its own current answers list.  The switch that fixes the empty strings though is going from question.responses(answer_sheet) to sheet_answers.  Responses method filters out empty string answers.  Once the answers array has the empty string answer properly, the rest of the method handles the cases correctly -- if a new answer is given, the empty string answer is marked for deletion, if an empty string answer is given, it recognizes there's already an answer with the empty string and does nothing.

**Comment by twinge** [comment]
:+1:

**Comment by andrewroth** [comment]
@twinge I ran the cap tests with the fe gem pointed to this branch, and found another bug, so this needs another look, will wait for :+1: again

**Comment by twinge** [comment]
:+1:

**Pull request merged**